wasm/jit/jit_amd64.go

```diff
@@ -807,27 +880,32 @@ func (b *amd64Builder) handleConstI64(o *wazeroir.OperationConstI64) error {
 	return nil
 }
 
-func (b *amd64Builder) moveStackToRegister(tp generalPurposeRegisterType, loc *valueLocation) error {
+func (b *amd64Builder) moveStackToRegisterWithAllocation(tp generalPurposeRegisterType, loc *valueLocation) error {
```

Comment:
vs splitting into `b.allocateRegisterAndMarkUsed(tp, loc)` followed by `b.moveStackToRegister(loc)`? This function maybe is doing too many things to say at once..

Reply:
haha yeah, I will leave as-is, but I will put some comments about this as TODO!

Reply:
https://github.com/tetratelabs/wazero/pull/71/commits/98fe7f43a3bcce6cbc91ae8887326c11bab710cb